drivers/entropy/CMakeLists.txt

```diff
@@ -1,10 +1,8 @@
-zephyr_library()
-
-zephyr_library_sources_ifdef(CONFIG_ENTROPY_ESP32_RNG          entropy_esp32.c)
-zephyr_library_sources_ifdef(CONFIG_ENTROPY_MCUX_RNGA          entropy_mcux_rnga.c)
-zephyr_library_sources_ifdef(CONFIG_ENTROPY_MCUX_TRNG          entropy_mcux_trng.c)
-zephyr_library_sources_ifdef(CONFIG_ENTROPY_NRF5_RNG           entropy_nrf5.c)
-zephyr_library_sources_ifdef(CONFIG_ENTROPY_SAM_RNG            entropy_sam.c)
-zephyr_library_sources_ifdef(CONFIG_ENTROPY_STM32_RNG          entropy_stm32.c)
-zephyr_library_sources_ifdef(CONFIG_FAKE_ENTROPY_NATIVE_POSIX  fake_entropy_native_posix.c)
-zephyr_library_sources_ifdef(CONFIG_USERSPACE                  entropy_handlers.c)
+zephyr_sources_ifdef(CONFIG_ENTROPY_ESP32_RNG          entropy_esp32.c)
+zephyr_sources_ifdef(CONFIG_ENTROPY_MCUX_RNGA          entropy_mcux_rnga.c)
+zephyr_sources_ifdef(CONFIG_ENTROPY_MCUX_TRNG          entropy_mcux_trng.c)
+zephyr_sources_ifdef(CONFIG_ENTROPY_NRF5_RNG           entropy_nrf5.c)
+zephyr_sources_ifdef(CONFIG_ENTROPY_SAM_RNG            entropy_sam.c)
+zephyr_sources_ifdef(CONFIG_ENTROPY_STM32_RNG          entropy_stm32.c)
+zephyr_sources_ifdef(CONFIG_FAKE_ENTROPY_NATIVE_POSIX  fake_entropy_native_posix.c)
+zephyr_sources_ifdef(CONFIG_USERSPACE                  entropy_handlers.c)
```

Comment:
I undertand that this scheme requires drivers to not use zephyr_library, but (albeit not for entropy) there are valid use-cases for using zephyr_library in drivers.

So we need to do something else.

I believe that if we merge #13791, then out-of-tree drivers can do

target_sources(drivers__entropy PRIVATE my/out/of/tree/driver.c)

Then we don't need to re-organize the drivers, and we don't need to avoid using zephyr_library_* in drivers.

Reply:
Sounds like that could work and is the sane way of doing this. I'll check out this locally and test if it works.

Reply:
Looks like the approach in #13791 works fine for this :+1: Anything stopping that from being merged?

Reply:
I'm certain it will be merged. It's blocked by review, review is going a bit slow these days due to embedded world.

Reply:
Ok, I removed the changes to the particular CMakeLists.txt file and we'll use the patch in #13791 in the out-of-source tree.